test/test_sspec_infrastructure.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+# Having trouble here - need to make sure we're getting
+# assertions and exceptions as expected.
+class TestSSpecInfrastructure < ShoesSpecLoggedTest
+  self.logger_dir = File.expand_path("#{__dir__}/../logger")
+
+  def test_simple_assertion_success
+    run_scarpe_sspec_code(<<~'SSPEC')
+      ---
+      ----------- app code
+      Shoes.app do
+      end
+      ----------- test code
+      assert_equal true, true
+    SSPEC
+  end
+
+  def test_exception
+    run_scarpe_sspec_code(<<~'SSPEC', expect_result: :error)
+      ---
+      ----------- app code
+      Shoes.app do
+      end
+      ----------- test code
+      raise "Yup, that's an exception!"
+    SSPEC
+  end
+
+  def test_many_assertions
+    run_scarpe_sspec_code(<<~'SSPEC', expect_assertions_min: 10)
+      ---
+      ----------- app code
+      Shoes.app do
+      end
+      ----------- test code
+      10.times { assert_equal true, true }
+    SSPEC
+  end
+
+  def test_skip
+    run_scarpe_sspec_code(<<~'SSPEC', expect_result: :skip)
+      ---
+      ----------- app code
+      Shoes.app do
+      end
+      ----------- test code
+      skip
+    SSPEC
+  end
+
+  def test_assertion_fail
+    run_scarpe_sspec_code(<<~'SSPEC', expect_result: :fail)
+      ---
```

Comment:
How should I implement this in the wild?

Define it as an .sspec file under the test directory?

Or am I able to write my examples/ code in this manner and it'll get pulled in for tests?

or do I need to run it in the "run_scarpe_sspec_code" block, as a ShoesSpecLoggedCase, in the test directory?

Reply:
Right now you'll need to run_scarpe_sspec. So we'll need an equivalent of test/test_examples.rb, where we go through a directory and call run_scarpe_sspec on each of them. Right now you'll *also* need to specify whether you expect the test to fail -- if something gets an error or a skip and you expect that, you'll need to tell it so. Eventually we should have frontmatter to specify that...

For Shoes-Spec I'm planning to add a rake task to do it. That'll make it easier to clone a Shoes-Spec dir somewhere and then run through it without requiring that the Shoes-Spec dir always be in the same place in the file system.

Reply:
> Right now you'll need to run_scarpe_sspec.

That's totally fine with me. More than happy to run that in an sspec block!

Reply:
It should catch on `rake test` if I want to add some?

Reply:
Again, you'd need a test_*.rb file to run it. But you could easily call run_scarpe_sspec on the file. As with test_examples, the best way to do a directory would be to list the files in there and make one Minitest test for each one. Something like:

```
class TestShoesSpecFiles < ShoesSpecLoggedTest
  self.logger_dir = File.expand_path("#{__dir__}/../logger")

  sspecs_to_test = Dir["spec/**/*.sspec"]
  sspecs_to_test.each.with_index do |filename, n|
    define_method("test_sspec_#{n}") do
      run_scarpe_sspec(filename)
    end
  end
end
```

I'm sure you could do better than that, but I'm just typing into a comment right now :-)

Reply:
If I do a rake task to run the Shoes-Spec files I'll do something similar -- find all the specs, find the results files, and write a Minitest test that expects it to pass, fail, error, skip, etc. according to what's in the results YAML. A little more complicated, but not ridiculously so.